lib/improver/utilities/cube_constraints.py

```diff
@@ -34,18 +34,25 @@
 import iris
 
 
-def create_sorted_lambda_constraint(coord_name, values):
+def create_sorted_lambda_constraint(coord_name, values, tolerance=1.0E-7):
     """
     Create a lambda constraint for a range. This formulation of specifying
     a lambda constraint has the benefit of not needing to hardcode the name
     for the coordinate, so that this can be determined at runtime.
 
+    The created function uses float values. As a result, a small tolerance is
+    used to spread the ends of the ranges to help with float equality
+    matching.
+
     Args:
         coord_name (str):
             Name of the coordinate.
         values (list):
             A list of two values that represent the inclusive end points
             of a range.
+        tolerance (float):
+            A relative tolerance value to ensure equivalence matching when
+            using float32 values.
```

Comment:
Possibly worth mentioning it won't help for zeroes - but I don't think we'd be using zero anywhere in SI units anyway (except for potentially cloud?).

Reply:
Hmm. I'd forgotten about zero. I've added explainer text as I cannot think of a sensible solution whilst we persist with precip rates in ms-1.